lightning/src/ln/onchain_utils.rs

```diff
@@ -0,0 +1,892 @@
+#[derive(PartialEq, Clone)]
+pub(crate) enum BumpStrategy {
+	RBF,
+	CPFP
+}
+
+impl Writeable for BumpStrategy {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), ::std::io::Error> {
+		match self {
+			BumpStrategy::RBF => {
+				writer.write_all(&[0; 1])?;
+			},
+			BumpStrategy::CPFP => {
+				writer.write_all(&[1; 1])?;
+			}
+		}
+		Ok(())
+	}
+}
+
+impl Readable for BumpStrategy {
+	fn read<R: ::std::io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let bump_strategy = match <u8 as Readable>::read(reader)? {
+			0 => {
+				BumpStrategy::RBF
+			},
+			1 => {
+				BumpStrategy::CPFP
+			},
+			_ => return Err(DecodeError::InvalidValue),
+		};
+		Ok(bump_strategy)
+	}
+}
+
+/// A structure to describe a claim content and its metadatas which is generated
+/// by ChannelMonitor and used by OnchainTxHandler to generate feerate-competive
+/// transactions.
+///
+/// Metadata are related to multiple fields playing a role in packet lifetime.
+/// Once issued, it may be aggregated with other requests if it's judged safe
+/// and feerate opportunistic.
+/// Current LN fees model, pre-committed fees with update_fee adjustement, means
+/// that counter-signed transactions must be CPFP to be dynamically confirmed as a
+/// bumping strategy. If transactions aren't lockdown (i.e justice transactions) we
+/// may RBF them.
+/// Feerate previous will serve as a feerate floor between different bumping attempts.
+/// Height timer clocks these different bumping attempts.
+/// Absolute timelock defines the block barrier at which claiming isn't exclusive
+/// to us anymore and thus we MUST have get it solved before.
+/// Height original serves as a packet timestamps to prune out claim in case of reorg.
+/// Content embeds transactions elements to generate transaction. See PackageTemplate.
+#[derive(PartialEq, Clone)]
+pub struct OnchainRequest {
+	// Timeout tx must have nLocktime set which means aggregating multiple
+	// ones must take the higher nLocktime among them to satisfy all of them.
+	// Sadly it has few pitfalls, a) it takes longuer to get fund back b) CLTV_DELTA
+	// of a sooner-HTLC could be swallowed by the highest nLocktime of the HTLC set.
+	// Do simplify we mark them as non-aggregable.
+	pub(crate) aggregation: bool,
+	// Content may lockdown with counter-signature of our counterparty
+	// or fully-malleable by our own. Depending on this bumping strategy
+	// must be adapted.
+	pub(crate) bump_strategy: BumpStrategy,
+	// Based feerate of previous broadcast. If resources available (either
+	// output value or utxo bumping).
+	pub(crate) feerate_previous: u64,
+	// At every block tick, used to check if pending claiming tx is taking too
+	// much time for confirmation and we need to bump it.
+	pub(crate) height_timer: Option<u32>,
+	// Block height before which claiming is exclusive to one party,
+	// after reaching it, claiming may be contentious.
+	pub(crate) absolute_timelock: u32,
```

Comment:
Same here, shouldn't this be a function on PackageTemplate?

Reply:
Yeah, I'll should study if I can merge OnchainRequest with PackageTemplate, now that I've isolated outpoints and their solving materials in PackageSolvingData.